src/schedule/MongoSchedule.ts

```diff
@@ -19,11 +25,23 @@ export class MongoSchedule extends Schedule {
    * @param connectionOptions for the MongoDB connection to establish
    */
   public static async connect(connectionOptions: MomoConnectionOptions): Promise<MongoSchedule> {
+    const connection = await Connection.create(connectionOptions);
+
+    const executionsRepository = connection.getExecutionsRepository();
+    const jobRepository = connection.getJobRepository();
+
     const scheduleId = uuid();
-    const mongoSchedule = new MongoSchedule(scheduleId);
-    await connect(connectionOptions);
-    await getExecutionsRepository().addSchedule(scheduleId);
+    await executionsRepository.addSchedule(scheduleId);
+
+    const mongoSchedule = new MongoSchedule(
+      scheduleId,
+      connection.disconnect.bind(connection),
```

Comment:
can we simplify here? Hand in the connection maybe (or will that make test setup worse?)

Reply:
The tests are using the static connect function to build instances. No changes required there.